Sources/PackageModel/Product.swift

```diff
@@ -69,6 +69,9 @@ public enum ProductType: Equatable {
 
     /// An executable product.
     case executable
+
+    /// An executable code snippet.
+    case snippet
```

Comment:
Did you consider making snippets a special kind of executable instead (e.g. by adding a type or something like that to executables)? 

I'm asking because as far as I can see, we mostly treat them the same, except for the purpose of listing them and only including them in the model for root packages, so it might be cleaner to not have to concern ourselves with the distinction everywhere. I am not entirely sure, though, maybe it *is* better to have a distinct type.

Reply:
I started out with that but it got to be a little too confusing for me to read. I also sort of expect that they will diverge more in the future.